platform/android/app/src/main/kotlin/coop/polypoly/polypod/homescreen/HomeScreenViewModel.kt

```diff
@@ -105,25 +109,49 @@ data class Footer(
 
 class HomeScreenViewModel {
     fun getSectionModels(
+        context: Context,
         onFeatureSelected: (String) -> Unit
     ): List<SectionModel> {
         return FeatureStorage.categories.map { category ->
             SectionModel(
                 category.name,
-                SectionType.fromCategoryType(category.category),
+                SectionType.fromCategoryType(category.id),
                 category.features.map { feature ->
                     TileModel(
                         feature.name,
-                        feature.description,
-                        feature.thumbnail,
+                        feature.description ?: "",
+                        createTileThumbnailBitmap(context, feature.thumbnail),
                         Color(feature.thumbnailColor),
                         Color(feature.borderColor),
-                        Color(feature.tileTextColor)
+                        Color(feature.tileTextColor),
                     ) {
                         onFeatureSelected(feature.id)
                     }
                 }
             )
         }
     }
+
+    private fun createTileThumbnailBitmap(
+        context: Context,
+        path: String?
+    ): Bitmap? {
+        if (path == null) {
+            return null
+        }
```

Comment:
how's this null going to be handled in the receiving end? Wouldn't it be simpler to just make it impossible to call without a valid string by eliminating the optionality?

Reply:
The function return type is Optional, and it is not optional only due to this - loading a bitmap can also return null. It would be simpler in this case, but can updated.